appcelerator/titanium_mobile
# fix(android)(9_3_X): build fails if "app_name" is defined in i18n "strings.xml" as of 9.0.0

**JIRA:**
https://jira.appcelerator.org/browse/TIMOB-28027

**Summary:**
As of Titanium 9.0.0, a build failure will occur if you have an `app_name` entry defined in a "strings.xml" file under the "i18n" folder. This will cause a resource name collision since Titanium's build system will add an `app_name` entry from the i18n "app.xml" or "tiapp.xml" file.

**Note:**
This regression is low priority because it's using an undocumented approach to localize the app name. The official cross-platform way of localize the app name is via the i18n "app.xml" file as described below.
https://docs.appcelerator.com/platform/latest/#!/guide/Internationalization-section-src-29004892_Internationalization-Appnamelocalization

**Test:**
1. Create a Classic Titanium app project.
2. Add the below i18n XML files to the project.
3. Build and run on Android.
4. Verify app builds and runs successfully. _(It used to fail.)_
5. Go to device's app list screen and verify name under icon is "app_name".
6. Delete the `app_name` line from the "strings.xml" file.
7. Build and run on Android.
8. Go to device's app list screen and verify name under icon is "AppXmlName".
9. Delete the "appname" line from the "app.xml" file.
10. Build and run on Android.
11. Go to app list screen and verify name under icon matches `<name/>` in "tiapp.xml".

./i18n/en/app.xml
```xml
<?xml version="1.0" encoding="UTF-8"?>
<resources>
	<string name="appname">AppXmlName</string>
</resources>
```

./i18n/en/strings.xml
```xml
<?xml version="1.0" encoding="UTF-8"?>
<resources>
	<string name="app_name">app_name</string>
	<string name="hello_world">Hello World</string>
</resources>
```
